tests/Unit/Evolution/Systems/GeneralizedHarmonic/TestFunctions.py

```diff
@@ -196,3 +196,332 @@ def two_index_constraint(d_gauge_function, spacetime_normal_one_form,
     return constraint
 
 # End test functions for two-index constraint
+
+# Test functions for F constraint
+
+
+def f_constraint_term_1_of_25(spacetime_normal_one_form,
+                              spacetime_normal_vector,
+                              inverse_spacetime_metric, d_pi):
+    spacetime_normal_vector_I = spacetime_normal_vector[1:]
+    d_pi_abc = np.pad(d_pi, ((1, 0), (0, 0), (0, 0)), 'constant')
+    term = np.einsum("bc,abc->a", inverse_spacetime_metric, d_pi_abc)
+    term += np.einsum("a,i,bc,ibc->a", spacetime_normal_one_form,
```

Comment:
Can you please clarify? Are you saying I can omit the "->a"? Even if I can, I prefer it, because it makes it clearer which indices are supposed to be free and which summed over when I read the code.

Reply:
Yes, you can omit the `->a`. In the code that follows you did that. I thought, if you omit it everywhere else, you should also do this here. It might be confusing if it is only used here.

Reply:
OK, I did this. I want to avoid adding precomputed temporaries as @hrueter suggested above for now, instead tackling performance in general for the constraints with an issue to be opened when this PR is merged.